API.md

```diff
@@ -0,0 +1,57 @@
+# Octofet API
+
+## `class Octofet`
+
+Uses: class Spi2Parallel
+
+Create an object of type `Octofet` to communicate with a particular [Octofet board](https://my.amperka.com/modules/octofet). The one board Octofet includes of 8 power switches.
+
+### `Octofet(uint8_t pinCS, uint8_t deviceCount = 1)`
+
+Constructs a new Octofet board object that uses the hardware SPI bus with arguments:
+
+- `pinCS`: pin Chip Select used for controls the shift-register latch. Any GPIO-pin can be used.
+- `deviceCount`: quantity of octofets boards connected in one chain: logical output pin headers of one module to the inputs of another. If the `deviceCount` argument is omitted, it used only one board Octofet.
+
+### `Octofet(uint8_t pinCS, uint8_t pinMOSI, uint8_t pinSCK, uint8_t deviceCount = 1)`
+
+Constructs a new Octofet board object that uses the software SPI bus with arguments:
+
+- `pinCS`: pin Chip Select used for controls the shift-register latch. Any GPIO-pin can be used.
+- `pinMOSI`: pin MOSI for sending data to the peripherals. Any GPIO-pin can be used.
+- `pinSCK`: pin SCK the clock pulses which synchronize data transmission generated by the master. Any GPIO-pin can be used.
+- `deviceCount`: quantity of octofets boards connected in one chain: logical output pin headers of one module to the inputs of another. If the `deviceCount` argument is omitted, it used only one board Octofet.
+
+Difference between hardware and software SPI is speed. Software SPI it will run much slower than hardware SPI, but hardware SPI works only on specific pins: `MOSI`, `MISO`(not used in Octofet) and `SCK`. Use software SPI only when hardware SPI pins are used by other devices.
+
+### `void begin()`
+
+Initializes the given interface, prepares the board for communication. All power switches are set to turn-off state.
+
+Important, call this method at least once, in your `setup()`.
+
+### `void digitalWrite(uint8_t switch, bool value, uint8_t device = 0)`
+
+Sets the state ("on" or "off") of one power switch. Function arguments:
+
+- `switch`: ordinal number power switch on device. Range is from `0` to `7`. Use option value is `ALL` to selected and sets state of all power switches.
+- `value`: states selected power switch. Values (turn-on or turn-off) may be switch is different data types: `HIGH` or `LOW` / `true` or `false` / `1` or `0`.
+- `device`: ordinal number of device on the bus. Range is from `0` to `n - 1`, where `n` is last number device. Use option value is `ALL` to selected and sets state of all power switches. If the `device` argument is omitted, it used only one board Octofet.
+
+### `void digitalWriteDevice(uint8_t value, uint8_t device = 0)`
```

Comment:
So, you suggest names:

```cpp
digitalWrite(..., device)
digitalWriteDevice(..., device)
```

Throw the context out of your head. What is the difference now? Both methods relate to a device somehow. So, the “Device” refinement suffix does not clearly communicate the difference. The main idea is writing $eight $channels at $once. Just say it. Suggest one of the following:

```
digitalWrite8   (like in OpenGL)
digitalWriteAllChannels   (verbose but straightforward)
digitalWriteAtOnce   (wordy but straightforward)
```